include/osquery/expected.h

```diff
@@ -0,0 +1,164 @@
+//
+//  expected.hpp
+//  gmock
+//
+//  Created by Max Kareta on 5/15/18.
+//
+
+#pragma once
+
+#include <include/osquery/error.h>
+
+/** Utility class that should be used in function that return
+ * either error or value. Expected enforce developer to test for success and
+ * check error if any.
+ *
+ * ExpectedValue:
+ * ExpectedValue<std::string> function() {
+ *   if (test) {
+ *    return "ok";
+ *   } else {
+ *    return std::shared_prt(new Error(error_domain, error_code));
```

Comment:
1. To avoid potential memory leak it is good to use `std::make_shared` and `std::make_unique` instead of using raw `new` operator. 
2. Misstype: `std::shared_ptr`

Reply:
Yes, this case is just a comment :)
But in case of using raw `new` to create of `shared_ptr` there is possibility to memory leak just because of exceptions. [link](https://stackoverflow.com/questions/20895648/difference-in-make-shared-and-normal-shared-ptr-in-c). IMHO, to be sure, is better to avoid using new, and create all smart pointers via `std::make_*`. Does it make sense?